agent/integration/main_test.go

```diff
@@ -0,0 +1,58 @@
+package integration
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/buildkite/agent/v3/clicommand"
+	"github.com/buildkite/agent/v3/version"
+	"github.com/urfave/cli"
+)
+
+var WriteExecutableCommand = cli.Command{
+	Name:  "write-exec",
+	Usage: "Write STDIN to an executable file",
+	Action: func(c *cli.Context) error {
+		path := c.Args().First()
+
+		f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o755)
```

Comment:
I wanted to use `os.Create` here, but it didn't take an argument to set the file permissions, which I need to be executble.

Reply:
That's reasonable. You could also call `f.Chmod(0o755)` separately.